### DISCUSSION TITLE
Onboarding Flow Ideas

### DISCUSSION THREAD

**Discussion by WillieCubed** [acmutd/comet_planning#49] [Ideas]
The maintainers have been talking about onboarding ideas, so I figured we should document them for future use.

As a reminder, there will be two "new setup" flows:
- New user onboarding, in which we get a student's profile/career information and save their preferences for the app
  - Triggered when a user opens the app for the first time on a device (disabled when a user has already gone through it)
- Degree plan creation, in which the user is guided through setting up their degree plan and is able to specify input courses.

@litiwu, @rajmeet2001, feel free to document what you had in mind here including assets/mockups/other ideas.

**Comment by rajmeet2001** [comment]
# Onboarding Questions

1. What’s your name? (Text Box)
2. Are you receiving financial student aid? (Yes/No)
3. What kind of scholarship are you receiving? (Dropdown of some common scholarships or an "other" section to write name and amount)
4. What are you studying? (Dropdown with choice to pick major and minor)
5. Are you part of any Honor Programs? (Drop Down)
5. Post Graduation Plans? (Med School, Law School, Grad School etc.)
6. Transfer Credit section (Adding questions based on the tool) (Option of AP/IB will be added)
7. What mode will you be taking? (3 Options) 
8. Do you plan on enrolling in the fast track program? (Yes/No)
9. What major will you be enrolling into as part of the fast track program? (Dropdown with choice to pick major)
10. Would you be interested in enrolling into courses during breaks (Summer, winter etc)? (Yes/No)

Here are some questions that are to be considered for the onboarding section. Please let me know if you have any questions that you would like to add, or change the styling of the questions presented. 

# Question & Comments I have

1. I think we should combine financial aid and scholarships under one section? I think it simplifies tracking the money. 
    We can differentiate between financial aid and scholarships using some tags or different colors.

**Reply by WillieCubed** [reply]
For questions with limited choices, could you enumerate all of them (like for post-graduation plans and related questions, specify all the possible options).
Additionally:
- For question 2 (financial aid), we should distinguish between scholarships and federal/state student aid to be more clear.
- Question 7 (transfer credit) should have have the same dialog as that which is used when generating an individual plan. It should be in a distinct section from the other user-specific questions.
- Questions 8, 9, 10 and 11 should be used only in the context of creating a new plan or modifying an existing one.

To answer your last question, feel free to group related questions into sections.

**Reply by rajmeet2001** [reply]
@WillieCubed I am a bit confused by your 3rd point. What do you mean by "only used in the context of creating a new plan"? Are we going to have them as a part of onboarding?

**Reply by WillieCubed** [reply]
We have user onboarding which is primarily run for the first time when we detect a user hasn't used Nebula Web before. We're distinguishing "User Onboarding" from "New Plan Creation." When we say "only used in the context of creating a new plan," I mean those questions are are a part of New Plan Creation, not User Onboarding, because they are more related to a specific plan instance when generating a sequence of coursework. We could have those questions as user-specific questions that are used as defaults, but for now, we can hold off on those and focus on the other questions in User Onboarding.

**Comment by litiwu** [comment]
![IMG_A956C201A285-1](https://user-images.githubusercontent.com/59784432/128291804-2ba487fb-1ab2-438b-a102-0f4186000f60.jpeg)

I updated the planning I had for the questions with the changes mentioned. 
The main updates are:
1. Enter the specific amount for financial aid received and select the type of financial aid
2. Add an option to select a specific major and enrollment time for fast-track programs.
3.  Incorporate AP/IB credits into the transfer credits dialogue.
4. Not including the "speed" of the plan and option to take summer classes in this portion of onboarding.

Things that I am still thinking about:
1. For post-graduation plans, how should the pre-professional options be presented, and should we separate masters and doctorates for graduate school options?
2. How to best create the ux for inputting IB and AP credits alongside other community college credits.

**Reply by wbne** [reply]
For question 1, master and doctorate applications are very similar if I recall correctly so we can probably put those 2 together and call it graduate school.

For question 2, I am thinking that we can have a drop down with the common transfer credit sources as options and clicking on one will make a list of common courses appear. The user would then check all the courses that they are going to transfer and courses not listed would probably link the transfer equivalency system.

**Comment by rajmeet2001** [comment]
# Onboarding Questions Cont. 

In our previous conversation, we discussed adding specific sections along with enumerating the choices for onboarding questions. Here are the changes:

1. **Basic Information/Student Data**
     - What’s your name? (Text Box)
     - Student Classification (Drop Down) (Freshman, Sophomore, Junior, Senior)
     - What is your major? (Drop Down)
     - (Optional) Second Major? (Drop Down)
     - (Optional) Minor? (Drop Down)
     - Will you be attending University Campus in-person? (Yes/No)
2. **Student Financial Aid**
     - Are you receiving any school-provided scholarships? (Yes/No)
     - Are you receiving federal financial student aid? (Yes/No)
     - If yes, what type of financial aid are you receiving ? (Drop Down)
 3. **Plans After Graduation**
      - What do you plan on doing after Graduation? (Multiple Choice) (Law School, Medical School, Research, Graduate School, Industry/Business , Military, Entrepreneurship, Undecided)
 4. **Honor Programs**
      - Are you part of any Honors Programs? (Drop Down) (Multiple Choice)
 5. **Transfer Section**
      - Much different from the others. We will have a list of "suggested courses" based on the student's year classification. 
      - Links to UTD Transfer Credit conversion tool, and AP Credits reference page.
      - Area to add these credits to the a credit portfolio. 
 6. **Modes and Tracks** (New Plan Creation)
      - Do you plan on enrolling/ is already in the fast track program? (Yes/No)
      - If yes, for what major? (Drop Down)
      - When did you enroll in the program or when will you plan on enrolling in the fast track program? (Drop Down)
      - Would you be interested in enrolling into courses during breaks (Summer, winter etc)? (Yes/No)
      - Finally, How fast do you want your college experience to go? (3 Modes) (Slow & Steady, Normal Pace, Fast & Furious) 

I know there has been some confusion on the Modes and Tracks section. @WillieCubed please let me know if something needs to be specified for that specific section.

**Reply by WillieCubed** [reply]
- What is the "Will you be attending University of Campus? (Yes/No)" question supposed to be asking?
- The "Are you receiving federal financial student aid?" question is too specific. The University provides a combination of federal, state, and other grants in student award packages. Rephrase it to reflect that fact. Additionally, put the question about scholarships before this question so it's clear from context there's a difference.
- There is no need to ask how much money students are actually receiving from the University because that does not affect plan generation. We're more interested in the fact they're on a specific scholarship like AES, National Merit, etc. (all of which are in the project's code) because those have GPA and credit hour requirements that affect coursework planning.
- Nice job listing the plans after graduation. I think each of those are good defaults. I would rename "Industry" to "Industry/Business" or something more specific. Additionally, graduate school may be too large of a category, but I think it's fine for now.
- Honors programs needs to be multiple choice because none of them are mutually exclusive. Someone can be a Collegium V, Liberal Arts Honors, Computer Science Scholar double-majoring in Economics and Computer Science, and all of those affect coursework planning.
- Section (6) should be tied to New Plan Creation, not User Onboarding (at least for now).

**Comment by ZzRanger** [comment]
# Onboarding UI
Currently, the new user onboarding is separated into 7 separate pages (3 for intro pages & 4 for main onboarding). I was thinking about possibly putting all the information on one interactive page in a manner similar to Typeform to enhance the user experience. What do you guys think about this?
